packages/core/src/accessanalyzer/vue/iamPolicyChecks.ts

```diff
@@ -0,0 +1,146 @@
+                        .getConfiguration()
+                        .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+                )
+            }
+        })
+    }
+
+    // TODO: Add functionality for the three checks
+    public async validatePolicy(documentType: string, policyType: string) {
+        this.client /* Do something */
+    }
+    public async checkNoNewAccess() {}
+    public async checkAccessNotGranted() {}
+}
+
+const Panel = VueWebview.compilePanel(IamPolicyChecksWebview)
+
+export async function renderIamPolicyChecks(context: ExtContext): Promise<void> {
+    const logger: Logger = getLogger()
+    try {
+        const client = new AccessAnalyzer({ region: context.regionProvider.defaultRegionId })
+        //Read from settings to auto-fill some inputs
+        const referencePolicyFilePath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.referencePolicyFilePath')!
+        const cfnParameterPath: string = vscode.workspace
+            .getConfiguration()
+            .get('aws.accessAnalyzer.policyChecks.cloudFormationParameterFilePath')!
+
+        //Check if Cfn and Tf tools are installed
+        let cfnToolInstalled = true
+        let tfToolInstalled = true
+        exec('tf-policy-validator', err => {
+            if (err) {
+                if (err.message.includes('command not found')) {
+                    tfToolInstalled = false
+                }
+            }
+        })
+        exec('cfn-policy-validator', err => {
+            if (err) {
+                if (err.message.includes('command not found')) {
+                    cfnToolInstalled = false
+                }
+            }
+        })
+
+        const wv = new Panel(
+            context.extensionContext,
+            {
+                referenceFilePath: referencePolicyFilePath ? referencePolicyFilePath : '',
+                cfnParameterPath: cfnParameterPath ? cfnParameterPath : '',
+                referenceDocument: await _getReferenceDocument(referencePolicyFilePath),
+                pythonToolsInstalled: cfnToolInstalled && tfToolInstalled,
+            },
+            client
+        )
+        await wv.show({
+            viewColumn: vscode.ViewColumn.Beside,
+            title: localize('AWS.iamPolicyChecks.title', 'IAM Policy Checks'),
+        })
+    } catch (err) {
+        logger.error(err as Error)
+    }
+}
+
+// Helper function to get document contents from a path
+async function _getReferenceDocument(input: string): Promise<string> {
+    if (fs.existsSync(input)) {
+        return fs.readFileSync(input).toString()
+    } else {
+        try {
+            const { region, bucket, key } = AmazonS3URI(input)
+            const s3Client = new S3({ region })
+            const resp = await s3Client.getObject({ Bucket: bucket, Key: key }).promise()
+            return resp.Body!.toString()
+        } catch (e: any) {}
```

Comment:
instead of taking a new dependency on a package for one usage, can we just use or add to util functions from https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/shared/clients/s3Client.ts ?

We should also use this to get the s3Client rather than constructing it here.

Reply:
I agree that it would be nice to not have a new dependency for one thing. Unfortunately, S3 as a service does not have a built-in way of extracting, or pulling, from an S3 URI(outside of Java SDK). This new dependency is an adoption of that Java implementation into Typescript. I can add a new helper function, but it will just be using the same algorithm as AmazonS3URI.

Reply:
I also construct the S3Client here because the region is variable depending on the provided link. I supposed that a new client with the appropriate region would need to be constructed every time.